lib/migration.js

```diff
@@ -295,12 +296,33 @@ function mixinMigration(PostgreSQL) {
     const self = this;
     const modelDef = this.getModelDefinition(model);
     const prop = modelDef.properties[propName];
+    let result = self.columnDataType(model, propName);
+
+    // checks if dataType is set to uuid
+    let postgDefaultFn;
+    let postgType;
+    const postgSettings = prop.postgresql;
+    if (postgSettings && postgSettings.dataType) {
+      postgType = postgSettings.dataType.toUpperCase();
+    }
+
     if (prop.generated) {
-      return 'SERIAL';
+      if (result === 'INTEGER') {
+        return 'SERIAL';
+      } else if (postgType === 'UUID') {
+        if (postgSettings && postgSettings.defaultFn && postgSettings.extension) {
```

Comment:
Hmm, how do we deal with the `extension`? Is it a function or just a boolean value?

Reply:
We need to 'install' the extension to use its functions. e.g, to use foo.bar(), we need to have foo installed first. In here extension is just a name that will be used in postgres query of creating table later.